packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js

```diff
@@ -121,6 +121,10 @@ if (!ShadowRoot.prototype.createElement) {
         promise = new Promise((r) => (resolve = r));
         this._definedPromises.set(tagName, promise);
         this._definedResolvers.set(tagName, resolve);
+        const definition = this._getDefinition(tagName);
```

Comment:
I think this check should go first: If the tag is defined, return `Promise.resolve(definition.elementClass)`, else store a Promise/resolver pair. Then we only have a single place to resolve the promise, and do any additional cleanup like in my other comment.

Nearby cleanup: think it'd be slightly nicer to have one map with an object value than two maps with the same key.

Reply:
I was assuming whenDefined should return the same promise for all calls with the same tagName; I just checked the [spec](https://html.spec.whatwg.org/multipage/custom-elements.html#dom-customelementregistry-whendefined) and it should but only until the element is defined, after that they get a fresh resolved promise. So I think that fits with what you suggested.

Made that change.